ext/page_log/palite/palite.cpp

```diff
@@ -26,6 +26,100 @@
+ *
+ * Each storage table type (Globals, Checkpoints, Pages) is implemented as a
+ * separate SQLite database file:
+ *  - globals.db
+ *  - checkpoints.db
+ *  - pages_[N].db  - one per WT table (N is the table ID)
+ *
+ * -= Concurrency model =-
+ *
+ * SQLite supports multiple simultaneous read transactions coming from separate
+ * database connections, possibly in separate threads or processes, but only one
+ * simultaneous write transaction.
+ *
+ * As a result, PALite uses a readers-writer lock at the table level to ensure
+ * that read operations can occur concurrently while write operations are
+ * serialized.
+ *
+ * -= Known Limitations =-
+ *
+ * PALite does not currently support multiple processes accessing the same
+ * database files. While SQLite supports this use case, it allows only one
+ * writer at a time. In absence of a proper locking mechanism between multiple
+ * processes at PALite level, multiple writers will conflict and fail.
+ * This limitation may be addressed in future releases.
+ *
+ * -= PALite structure =-
+ *
+ * The following diagram illustrates the overall structure of PALite:
+ *
+ * -----------------------------------------------------------------------------
+ *
+ * PALI implementation:
+ *
+ * PageLog
+ *   |
+ *   +- [PageHandle] - list: an instance per WT table
+ *   +- Config - shared by all components
+ *   +- Storage - shared by all page handles
+ *
+ * - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+ *
+ * Storage public methods:
+ *
+ *   - make_next_lsn   - get/put_page         - discard_page   - abandon_checkpoint >--+
+ *   - get_last_lsn    - get/put_checkpoint   - get_page_ids   - ...                   |
+ *                                                                                     |
+ * Everything at this level calls public methods of the underlying tables.             |
+ * Internally these methods will use appropriate R/W lock for the call.                |
+ * The only exception is 'abandon_checkpoint' which bypasses the locks completely.     |
+ *                                                                                     |
+ * - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -       |
+ *                                                                                     |
+ *                                                                | bypasses per-table access; |
+ *                                                                | uses storage-wide lock     |
+ *                                                                                     |
+ *       < - - - - - - - - - - - - - - - - - - - - - - - - - - - - - >   <-------------+
+ *       <   next_lsn         |   get/put          |   get/put       >
+ *       <   last_lsn         |   delete           |   discard       > - specific for each class
+ *       <   ...              |                    |   ...           >
```

Comment:
Shouldn't this table classes specific methods be mentioned after mentioning the table classes?


```suggestion
 * Table <   Globals         |   Checkpoints      |     Pages       > - classes for tables
 *       <   next_lsn        |   get/put          |   get/put       >
 *       <   last_lsn        |   delete           |   discard       > - specific for each class
 *       <   ...             |                    |   ...           >
```